File: packages/gamole_types/schemas/linear.py
"""Linear schemas - ported from packages/types/src/schemas/linear.ts."""

from typing import Literal

from pydantic import BaseModel, Field


class LinearPushConfig(BaseModel):
    team_id: str | None = Field(default=None, alias="teamId")
    project_id: str | None = Field(default=None, alias="projectId")
    labels: list[str] | None = None
    state_id: str | None = Field(default=None, alias="stateId")
    mode: Literal["create", "update"] = "create"
    existing_issue_id: str | None = Field(default=None, alias="existingIssueId")

    model_config = {"populate_by_name": True}


class CreatedIssueInfo(BaseModel):
    linear_id: str = Field(alias="linearId")
    identifier: str
    title: str

    model_config = {"populate_by_name": True}


class CreatedRelationInfo(BaseModel):
    id: str
    type: str


class LinearPushResult(BaseModel):
    created_issues: list[CreatedIssueInfo] = Field(alias="createdIssues")
    created_relations: list[CreatedRelationInfo] = Field(alias="createdRelations")
    errors: list[str]

    model_config = {"populate_by_name": True}

File: packages/types/src/schemas/linear.ts
import { z } from 'zod';

// LinearPushConfig schema
export const LinearPushConfigSchema = z.object({
  teamId: z.string(),
  projectId: z.string().optional(),
  labels: z.array(z.string()).optional(),
  stateId: z.string().optional(),
  mode: z.enum(['create', 'update']).default('create'),
  existingIssueId: z.string().optional(),
});

export type LinearPushConfig = z.infer<typeof LinearPushConfigSchema>;

// LinearPushResult schema
export const LinearPushResultSchema = z.object({
  createdIssues: z.array(
    z.object({
      linearId: z.string(),
      identifier: z.string(),
      title: z.string(),
    })
  ),
  createdRelations: z.array(
    z.object({
      id: z.string(),
      type: z.string(),
    })
  ),
  errors: z.array(z.string()),
});

export type LinearPushResult = z.infer<typeof LinearPushResultSchema>;
